csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs

```diff
@@ -75,6 +75,30 @@ public static SessionOptions MakeSessionOptionWithCudaProvider(int deviceId = 0)
             return options;
         }
 
+        /// <summary>
+        /// A helper method to construct a SessionOptions object for CUDA execution.
+        /// Use only if CUDA is installed and you have the onnxruntime package specific to this Execution Provider.
+        /// </summary>
+        /// <param name="cuda_options">CUDA EP provider options to configure the CUDA EP instance</param>>
+        /// <returns>A SessionsOptions() object configured for execution with CUDA provider options.</returns>
+        public static SessionOptions MakeSessionOptionWithCudaProvider(OrtCUDAProviderOptions cuda_options)
+        {
+            CheckCudaExecutionProviderDLLs();
+            SessionOptions options = new SessionOptions();
+
+            OrtCUDAProviderOptionsNative cuda_options_native;
```

Comment:
>cuda_options_native [](start = 41, length = 19)

Where is the code that disposes of cuda_options_native?

Reply:
cuda_options_native is managed object, so it will be reclaimed by gc. Correct me if I'm wrong.

Reply:
Sure it is. Why is it called native then when it is managed?

---
In reply to: [612567480](https://github.com/microsoft/onnxruntime/pull/7315#discussion_r612567480) [](ancestors = 612567480)

Reply:
Rename it to avoid confusion.